Documentation/observability/metrics.rst

```diff
@@ -1271,3 +1271,24 @@ which is passed as the ``controller-group-metrics`` configuration
 flag. The current default set for ``kvstoremesh`` found in the
 Cilium Helm chart is the special name "all", which enables the metric
 for all controller groups. The special name "none" is also supported.
+
+NAT
+~~~
+
+======================================== ================================================== ========== ========================================================
+Name                                     Labels                                             Default    Description
+======================================== ================================================== ========== ========================================================
+``nat_endpoint_max_connection``          ``family``                                         Enabled    Saturation of the most the most saturated distinct NAT mapped connection, in terms of egress-IP and remote endpoint address.
+======================================== ================================================== ========== ========================================================
+
+These metrics are for monitoring Cilium's NAT mapping functionality. NAT is used by features such as Egress Gateway and BPF masquerading.
+
+The NAT map holds mappings for masqueraded connections. Connection held in the NAT table that are masqueraded with the
+same egress-IP and are going to the same remote endpoint address all require a unique source port for the mapping.
```

Comment:
Can we clarify what "remote endpoint" means? I assume it's typically IP (v4|v6) address + (TCP|UDP) port.

Maybe to add it here:

> Connection held in the NAT table that are masqueraded with the same egress-IP and are going to the same remote endpoint address all require a unique source port for the mapping.

Could we mention the destination port explicitly too?

> [...] going to the same remote endpoint (address and port) [...]